tests/reporter/kubernetes.go

```diff
@@ -337,6 +350,133 @@ func (r *KubernetesReporter) logAuditLogs(virtCli kubecli.KubevirtClient, logsdi
 	}
 }
 
+func (r *KubernetesReporter) logVMICommands(virtCli kubecli.KubevirtClient, isExternal bool) {
+
+	logsdir := filepath.Join(r.artifactsDir, "network", "vmis")
+	if err := os.MkdirAll(logsdir, 0777); err != nil {
+		fmt.Fprintf(os.Stderr, "failed to create directory: %v\n", err)
+		return
+	}
+
+	vmiNamespaces := tests.TestNamespaces
+	if isExternal {
+		vmiNamespaces = []string{v1.NamespaceAll}
+	}
+
+	for _, ns := range vmiNamespaces {
+		vmis, err := virtCli.VirtualMachineInstance(ns).List(&metav1.ListOptions{})
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "failed to get vmis: %v\n", err)
+			continue
+		}
+
+		for _, vmi := range vmis.Items {
+			if vmi.Status.Phase != "Running" {
+				fmt.Fprintf(os.Stderr, "skipping vmi %s, phase is not Running\n", vmi.ObjectMeta.Name)
+				continue
+			}
+
+			vmiType, err := getVmiType(vmi)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "failed get vmi type: %v\n", err)
+				continue
+			}
+
+			if err := prepareVmiConsole(vmi, vmiType); err != nil {
+				fmt.Fprintf(os.Stderr, "failed login to vmi: %v\n", err)
+				continue
+			}
+
+			r.executeVMICommands(vmi, logsdir, vmiType)
+		}
+	}
+}
+
+func (r *KubernetesReporter) logVirtLauncherCommands(virtCli kubecli.KubevirtClient, virtHandlerPods *v1.PodList) {
+
+	logsdir := filepath.Join(r.artifactsDir, "network", "pods")
+	if err := os.MkdirAll(logsdir, 0777); err != nil {
+		fmt.Fprintf(os.Stderr, "failed to create directory: %v\n", err)
+		return
+	}
+
+	virtLauncherPods, err := virtCli.CoreV1().Pods(v1.NamespaceAll).List(context.Background(), metav1.ListOptions{LabelSelector: fmt.Sprintf("%s=virt-launcher", v12.AppLabel)})
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "failed to fetch pods: %v\n", err)
+		return
+	}
+
+	for _, pod := range virtLauncherPods.Items {
+		if pod.Status.Phase != "Running" {
+			fmt.Fprintf(os.Stderr, "skipping pod %s, phase is not Running\n", pod.ObjectMeta.Name)
+			continue
+		}
+
+		if found := podHasComputeContainer(pod); found != true {
+			fmt.Fprintf(os.Stderr, "could not find compute container for pod %s\n", pod.ObjectMeta.Name)
+			continue
+		}
+
+		r.executeVirtLauncherCommands(virtCli, logsdir, pod)
+	}
+
+	for _, virtHandlerPod := range virtHandlerPods.Items {
+		if virtHandlerPod.Status.Phase != "Running" {
+			fmt.Fprintf(os.Stderr, "skipping virt-handler %s, phase is not Running\n", virtHandlerPod.ObjectMeta.Name)
+			continue
+		}
+
+		processList, err := getProcessList(virtCli, &virtHandlerPod)
+		if err != nil {
+			continue
+		}
+
+		lines := strings.Split(processList, "\n")
```

Comment:
This is what i tried to do with `parseProcessLine` (line 437) and line (429), so please elaborate
lets discuss offline about all the points here if possible (this one and https://github.com/kubevirt/kubevirt/pull/6116/files#r681184802)
since they are related to each other and have many small details